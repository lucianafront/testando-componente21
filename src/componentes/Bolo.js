import React from 'react';

function Bolo(props) {
	return (
		<div className="col-md-3 col-6 todosindredientes avela cenoura nutella">
			<div className="bolo-box center">
				<span>#{props.bolo.numero}</span>
				<a href= {props.bolo.href} target="_blank" rel="noreferrer">
					<img loading="lazy" width="235" height="235"
						src= {props.bolo.src}
						className="attachment-full size-full wp-post-image" alt="" decoding="async"
						srcSet= {props.bolo.srcSet}
						sizes="(max-width: 235px) 100vw, 235px" />
				</a>
				<h2>
					<a href= {props.bolo.href} target="_blank" rel="noreferrer">
						{props.bolo.nome}
					</a>
				</h2>
				<a href= {props.bolo.href}
					className="btn-bolos" target="_blank" rel="noreferrer">
					Eu quero
				</a>
			</div>
			<br />
			<br />
		</div>
	);
};

export default Bolo;