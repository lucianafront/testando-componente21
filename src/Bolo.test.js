import React, { act } from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Bolo from "./componentes/Bolo";

test("deve renderizar o componente Bolo e verificar interações", async () => {
  const bolo = {
    numero: 1,
    href: "https://example.com",
    src: "https://example.com/image.jpg",
    srcSet:
      "https://example.com/image.jpg 1x, https://example.com/image@2x.jpg 2x",
    nome: "Bolo de Cenoura",
  };

  await act(() => {
    render(<Bolo bolo={bolo} />);
  });

  // Verificar se o número do bolo é renderizado
  expect(screen.getByText("#1")).toBeInTheDocument();

  // Verificar se o nome do bolo é renderizado
  expect(screen.getByText("Bolo de Cenoura")).toBeInTheDocument();

  // Simular clique no link "Eu quero"
  await act(async () => {
    userEvent.click(screen.getByText("Eu quero"));
  });

  // Verificar se o link foi clicado (pode ser necessário mockar o comportamento)
  expect(screen.getByRole("link", { name: /eu quero/i })).toHaveAttribute(
    "href",
    "https://example.com"
  );
});

test("deve corresponder ao snapshot do componente Bolo", () => {
  const bolo = {
    numero: 1,
    href: "https://example.com",
    src: "https://example.com/image.jpg",
    srcSet:
      "https://example.com/image.jpg 1x, https://example.com/image@2x.jpg 2x",
    nome: "Bolo de Cenoura",
  };

  const { container } = render(<Bolo bolo={bolo} />);

  expect(container).toMatchSnapshot();
});

test("deve verificar se o componente Bolo exibe a imagem correta", () => {
  const bolo = {
    numero: 1,
    href: "https://example.com",
    src: "https://example.com/image.jpg",
    srcSet:
      "https://example.com/image.jpg 1x, https://example.com/image@2x.jpg 2x",
    nome: "Bolo de Cenoura",
  };

  render(<Bolo bolo={bolo} />);

  // Verificar se a imagem é renderizada corretamente
  const imageElement = screen.getByRole("img");
  expect(imageElement).toBeInTheDocument();
  expect(imageElement).toHaveAttribute("src", "https://example.com/image.jpg");
  expect(imageElement).toHaveAttribute(
    "srcSet",
    "https://example.com/image.jpg 1x, https://example.com/image@2x.jpg 2x"
  );
});


test("deve lançar um erro se as props obrigatórias não forem fornecidas", () => {
  // Render the Bolo component without providing the required props
  expect(() => {
    render(<Bolo />);
  }).toThrow();
});
