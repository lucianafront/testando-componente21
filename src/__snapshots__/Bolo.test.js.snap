// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deve corresponder ao snapshot do componente Bolo 1`] = `
<div>
  <div
    class="col-md-3 col-6 todosindredientes avela cenoura nutella"
  >
    <div
      class="bolo-box center"
    >
      <span>
        #
        1
      </span>
      <a
        href="https://example.com"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt=""
          class="attachment-full size-full wp-post-image"
          decoding="async"
          height="235"
          loading="lazy"
          sizes="(max-width: 235px) 100vw, 235px"
          src="https://example.com/image.jpg"
          srcset="https://example.com/image.jpg 1x, https://example.com/image@2x.jpg 2x"
          width="235"
        />
      </a>
      <h2>
        <a
          href="https://example.com"
          rel="noreferrer"
          target="_blank"
        >
          Bolo de Cenoura
        </a>
      </h2>
      <a
        class="btn-bolos"
        href="https://example.com"
        rel="noreferrer"
        target="_blank"
      >
        Eu quero
      </a>
    </div>
    <br />
    <br />
  </div>
</div>
`;
